### ISSUE TITLE
v0.4.0 Effects Streaming Api is timing out

### ISSUE THREAD

**Issue by dabitdev** [stellar/java-stellar-sdk#163]
I am just testing the streaming under EffectsRequestBuilder and I am getting timeout.
Is the streaming api using the read timeout set in the okhttpclient?

```
2018-12-17 19:27:37.558 2863-2934/com.blockeq.stellarwallet E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
    Process: com.blockeq.stellarwallet, PID: 2863
    java.lang.IllegalStateException: Failed 200
        at org.stellar.sdk.requests.SSEStream$StellarEventSourceListener.onFailure(SSEStream.java:174)
        at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:83)
        at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54)
        at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153)
        at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.net.SocketTimeoutException: timeout
        at shadow.okio.Okio$4.newTimeoutException(Okio.java:232)
        at shadow.okio.AsyncTimeout.exit(AsyncTimeout.java:285)
        at shadow.okio.AsyncTimeout$2.read(AsyncTimeout.java:241)
        at shadow.okio.RealBufferedSource.request(RealBufferedSource.java:68)
        at shadow.okio.RealBufferedSource.require(RealBufferedSource.java:61)
        at shadow.okio.RealBufferedSource.readHexadecimalUnsignedLong(RealBufferedSource.java:304)
        at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.readChunkSize(Http1Codec.java:469)
        at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.read(Http1Codec.java:449)
        at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:394)
        at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:383)
        at shadow.okhttp3.internal.sse.ServerSentEventReader.processNextEvent(ServerSentEventReader.java:61)
        at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:80)
        at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54) 
        at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153) 
        at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
     Caused by: java.net.SocketException: socket is closed
        at com.android.org.conscrypt.ConscryptFileDescriptorSocket$SSLInputStream.read(ConscryptFileDescriptorSocket.java:551)
        at shadow.okio.Okio$2.read(Okio.java:140)
        at shadow.okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
        at shadow.okio.RealBufferedSource.request(RealBufferedSource.java:68) 
        at shadow.okio.RealBufferedSource.require(RealBufferedSource.java:61) 
        at shadow.okio.RealBufferedSource.readHexadecimalUnsignedLong(RealBufferedSource.java:304) 
        at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.readChunkSize(Http1Codec.java:469) 
        at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.read(Http1Codec.java:449) 
        at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:394) 
        at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:383) 
        at shadow.okhttp3.internal.sse.ServerSentEventReader.processNextEvent(ServerSentEventReader.java:61) 
        at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:80) 
        at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54) 
        at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153) 
        at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
```

**Comment by xkf4806** [comment]
ah,I got the same error info while using PaymentsRequestBuilder...

**Comment by ChrisLMills** [comment]
Me too...

2018-12-19 21:50:58.939 17883-17943/? E/Fabric: Settings request failed.
    io.fabric.sdk.android.services.network.HttpRequest$HttpRequestException: java.net.SocketTimeoutException: connect timed out
        at io.fabric.sdk.android.services.network.HttpRequest.code(HttpRequest.java:1355)
        at io.fabric.sdk.android.services.settings.DefaultSettingsSpiCall.handleResponse(DefaultSettingsSpiCall.java:106)
        at io.fabric.sdk.android.services.settings.DefaultSettingsSpiCall.invoke(DefaultSettingsSpiCall.java:90)
        at io.fabric.sdk.android.services.settings.DefaultSettingsController.loadSettingsData(DefaultSettingsController.java:80)
        at io.fabric.sdk.android.services.settings.DefaultSettingsController.loadSettingsData(DefaultSettingsController.java:64)
        at io.fabric.sdk.android.services.settings.Settings.loadSettingsData(Settings.java:155)
        at io.fabric.sdk.android.Onboarding.retrieveSettingsData(Onboarding.java:126)
        at io.fabric.sdk.android.Onboarding.doInBackground(Onboarding.java:99)
        at io.fabric.sdk.android.Onboarding.doInBackground(Onboarding.java:45)
        at io.fabric.sdk.android.InitializationTask.doInBackground(InitializationTask.java:63)
        at io.fabric.sdk.android.InitializationTask.doInBackground(InitializationTask.java:28)
        at io.fabric.sdk.android.services.concurrency.AsyncTask$2.call(AsyncTask.java:311)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:762)
     Caused by: java.net.SocketTimeoutException: connect timed out
        at java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:334)
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:196)
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:178)
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:356)
        at java.net.Socket.connect(Socket.java:586)
        at com.android.okhttp.internal.Platform.connectSocket(Platform.java:113)
        at com.android.okhttp.Connection.connectSocket(Connection.java:1455)
        at com.android.okhttp.Connection.connect(Connection.java:1413)
        at com.android.okhttp.Connection.connectAndSetOwner(Connection.java:1700)
        at com.android.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:133)
        at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:466)
        at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:371)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:503)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:438)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:567)
        at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:105)
        at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java)
        at com.google.firebase.perf.network.zze.getResponseCode(Unknown Source)
        at com.google.firebase.perf.network.zzd.getResponseCode(Unknown Source)
        at io.fabric.sdk.android.services.network.HttpRequest.code(HttpRequest.java:1353)
        at io.fabric.sdk.android.services.settings.DefaultSettingsSpiCall.handleResponse(DefaultSettingsSpiCall.java:106) 
        at io.fabric.sdk.android.services.settings.DefaultSettingsSpiCall.invoke(DefaultSettingsSpiCall.java:90) 
        at io.fabric.sdk.android.services.settings.DefaultSettingsController.loadSettingsData(DefaultSettingsController.java:80) 
        at io.fabric.sdk.android.services.settings.DefaultSettingsController.loadSettingsData(DefaultSettingsController.java:64) 
        at io.fabric.sdk.android.services.settings.Settings.loadSettingsData(Settings.java:155) 
        at io.fabric.sdk.android.Onboarding.retrieveSettingsData(Onboarding.java:126) 
        at io.fabric.sdk.android.Onboarding.doInBackground(Onboarding.java:99) 
        at io.fabric.sdk.android.Onboarding.doInBackground(Onboarding.java:45) 
        at io.fabric.sdk.android.InitializationTask.doInBackground(InitializationTask.java:63) 
        at io.fabric.sdk.android.InitializationTask.doInBackground(InitializationTask.java:28) 
        at io.fabric.sdk.android.services.concurrency.AsyncTask$2.call(AsyncTask.java:311) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:237) 
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:237) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
        at java.lang.Thread.run(Thread.java:762)

**Comment by bartekn** [comment]
Thanks for reports. Fixed in [0.4.1](https://github.com/stellar/java-stellar-sdk/releases/tag/0.4.1).

**Comment by ChrisLMills** [comment]
Thanks for the quick response, it's working for me now.
